GoofEvents: Buffs some under-used events up

Iamgoofball opened the pull request:
# Lights Out
Makes it blow out a lot more lights than just one solitary room. Better grab a flashlight or a Janitor.

# Invasion Of The Mice
Hahahaha, this is so useless with only 5 mice usually. Buffs the amount spawned, Engineers better be on their guard.

# THEY'RE COMING OUT OF THE CLOGGED VENTS
![the-blob-14795036951](https://cloud.githubusercontent.com/assets/4081722/23839765/923512b0-075d-11e7-998f-7072a1e62811.jpg)

Be prepared.

Iamgoofball committed:
GoofEvents: Buffs some under-used events up

Iamgoofball committed:
ssssss

Kromgar commented:
Lights out takes out a couple rooms the fuck you talkin aboot

chanoc1 commented:
im in support of this pr

this gets the chanoc seal of approval

Iamgoofball commented:
@Kromgar Not as many as it does now, and it's still not a problem.

Poojawa commented:
Make sure you have some sort of blacklist for stuff like blob chems, or stuff that'll melt through the station.

otherwise, wew. lord.

Iamgoofball commented:
@Poojawa 
By default the proc for pulling random IDs specifically blacklists shit that can't be synth'd like blob chems and viruses

Poojawa commented:
Neat. all for it then. Should be hella fun.

optimumtact commented on code/modules/events/electrical_storm.dm:
```diff
@@ -17,20 +16,20 @@
-			epicentreList += pick(possibleEpicentres)
-		else
-			break
+	for(var/obj/machinery/power/apc/apc in world)
```
teh fuck

optimumtact replied:
if your'e goign to bookkeep a list of all APC's make it a global list updated on intialize/destroy

Iamgoofball replied:
do we have an apc list

optimumtact replied:
no, make one

Iamgoofball replied:
k

Shadowlight213 replied:
We have an apc list. Check the grid_check event.

Cyberboss replied:
or the machines list

optimumtact commented on code/modules/events/vent_clog.dm:
```diff
@@ -20,30 +17,23 @@
+	for(var/obj/machinery/atmospherics/components/unary/vent in vents)
 		if(vent && vent.loc)
-			var/datum/reagents/R = new/datum/reagents(50)
+			var/datum/reagents/R = new/datum/reagents(1000)
```
this is literally 20 times more reagent

Iamgoofball replied:
yes

Iamgoofball replied:
it needed a buff

coiax replied:
It's also foam, rather than smoke.

Iamgoofball replied:
Yup

ma44 commented:
>Picks a random reagent
>Spawns 1,000 units of it
>Also heats it up

So what happens if it spawns some meth or nitroglycerin?

Iamgoofball commented:
It doesn't heat it up tho?

octareenroon91 commented:
Note the vents event has been changed from smoke to foam.

ShadowDeath6 commented on code/game/objects/effects/effect_system/effects_foam.dm:
```diff
@@ -128,6 +128,9 @@
 	return
 
+/obj/effect/particle_effect/foam/long_life
+	lifetime = 150
```
this is in deciseconds, right?

Iamgoofball replied:
Yeah

MMMiracles commented:
>50 mice

oh god no its bad enough when one mouse chews a random wire in maintenance let alone 50 to eat the entire maintenance's wiring system.

YakumoChen commented:
5 was more then enough, goof.

GunHog commented on code/modules/events/vent_clog.dm:
```diff
@@ -20,30 +17,23 @@
+			var/datum/reagents/R = new/datum/reagents(1000)
 			R.my_atom = vent
-			R.add_reagent(pick(gunk), 50)
+			R.add_reagent(pick(get_random_reagent_id()), 1000)
```
Do you not think that this might be just a _little_ extreme? 1,000 units of clf3, acid, and other powerful random chems that could take down entire departments at once? Pick one: More units of a gunk chem, or 50 units of the expanded chem list.

Iamgoofball replied:
Think for a moment exactly how many chems we have

coiax replied:
Keep the gunk list, and have a small chance of picking a different random reagent.

Otherwise, it'll just be harmless booze coming out of the vents.

Iamgoofball replied:
Nope. Point is that gunk list is shit.

octareenroon91 replied:
OK look. As it stands, `get_random_reagent_id()`* includes
 * all consumables; 
 * all the species mutation toxins (most notably including plasmaman mutation toxin);
 * yes **clf3**.

Very little is actually blacklisted with `can_synth = 0`.

In code/modules/reagents/chemistry/holder.dm .

coiax commented:
Also, split your PR plz

Supermichael777 commented:
1000 units of booze is deadly and requires competent chem

Repukan commented:
Just inject charcoal, antihol or calomel?

Iamgoofball commented:
This isGoofevents

ma44 commented:
Even with all the reagents, this is literally just a "oh hey 10% of the time this even does nothing", the times it does happen kills departments or people

XDTM commented:
When ma44 makes a valid criticism you know your PR has balance issues

Niknakflak commented:
Its 5 to 15 mice currently.  5 is maybe too little but 15 is definitely where it should be.
25 to 50 mice is fucking insane.  Thumbs downing this PR for that mostly.
Don't really know how the light thing compares.
The foam thing I also really don't know because instead of explaining your PR, you put a meme image.
The description is pretty vague.

Iamgoofball commented:
I agree with you on the mice, we should probably do some testing though

Bluespace commented:
20 mice is a good spot.
Making clogged vents have a chance to fuck you up is good, gives medical a bit more to do. 
"SIR WHAT CHEM DID YOU INHALE"
"IT WAS BROWN AND MELTED MY SHOES OFF"

FantasticFwoosh commented:
Goof feature around events = biblical plague of egypt.

Supermichael777 commented:
mint toxin dosnt even work anymore

PKPenguin321 commented:
I was kinda up for this but these actual changes aren't any good
Lights breaking is already a bitch and a half, too many mice might as well be replaced with a powersink event, foam/more chems from vents is cool but now it can blow up entire departments/shoot romerol

Iamgoofball commented:
Romerol is not synthable and therefore isn't in the pool

PKPenguin321 commented:
Congrats, you've defeated one sub-example out of my three arguments

optimumtact commented:
I play pot of greed, it lets me draw two cards from my deck

Cheridan commented:
Events should be fun or engaging to respond to

Half the station suddenly has blown lights: Not fun to respond to. Would be better if it was just a localized power outage or something and they came on after a period.
50 mice in maint: Not fun, we shouldn't expect people to play whack-a-mole like this just so the station doesn't lose power.
FOAM PARTY: Sounds fun; ship it.

Bawhoppen commented:
@cheridan what's fun?

Iamgoofball commented:
@Cheridan would these changes be acceptable:
light burst: switched to be a department ala greytide virus
mice: dropped to 15-20 like NikNak suggested
FOAM: foam

PKPenguin321 commented:
im cheridan and yeah do that

Shadowlight213 commented:
Add an admin button to edit foam amount/range so we can cover the entire station in foam.

Iamgoofball commented:
@Shadowlight213 I might do that for another PR reworking admin events entirely because ATM it'd be snowflake AF

Iamgoofball commented:
but thats a good idea

octareenroon91 commented:
Foam party best party.

Colorful reagent

Iamgoofball commented:
whered the downvote storm come from lmao did the armchair designers show up?

CPTANT commented:
Wait doesn't random chemicals mean 1000 units of nitroglycerin could spawn in every vent?

Wouldn't that instantly maxcap every vent on the station?

XDTM commented:
If nitroglycerin spawns as a reagent it shouldn't explode. Explosions happen on reactions, or require above-average temperature.

ma44 commented:
Note that this also spawns the reagents literally at every vent possible, meaning facid, sacid or other stuff can permanently kill a large part of the station with breaches or dead bodies.

Iamgoofball commented:
Yup.

kevinz000 commented:
Therefore goofball is once again an idiot

Iamgoofball commented:
It's like it's supposed to be dangerous or something

I'm probably going to make it so that if you climb on a table you can avoid the foam, and provide earlier warning before the spew

Iamgoofball commented:
also we have over 800+ chems meaning the chance for acid is 2/800

Iamgoofball commented:
this is less deadly than the original plan to make it spew a random gas so

CPTANT commented:
Ohw well, why not

ShadowDeath6 commented:
1000u of 15 second foam from every vent is probably deadly more than not because ODing on things as well as deadly shit like mutation toxins that can be made

You probably need to make people able to dodge the foam in this pr, the foam event is probably going to be lethal on a large scale more often than not as there are very few chems that are both non lethal and don't OD

octareenroon91 commented:
Can we merge/testmerge foam event an alternative 0-weighted event?

Repukan commented:
>You probably need to make people able to dodge the foam 

You do have a way to dodge, you get almost 20 seconds from the announcement to move out of the way

octareenroon91 commented:
With the foam, especially buffed foam, you would need to get **_far away from every scrubber._**

Iamgoofball commented:
Foam stays as is. I'll nerf the other two events in a few hours.

kevinz000 commented:
Five upvotes sixteen downvotes gg

YakumoChen commented:
It's almost like people don't want every single event to force a shuttle call????

octareenroon91 commented:
@Iamgoofball did you have those tweaks ready for posting yet?

Iamgoofball commented:
got sidetracked by planetstation

@YakumoChen go back to bay

YakumoChen commented:
That doesn't sound like you're refuting my point.

Iamgoofball commented:
Why should I refute a point of you not being able to handle stubbing your toe?

YakumoChen commented:
Why should the station be forced to deal with an event that decimates the station after only a few minutes? Players are lazy and are going to yell at engineers to fix things like 50 mice in maint chewing the entire power system apart, captain is just going to force a shuttle call when he gets fed up with polyacid foam killing half the population. Stuff like meteors already is an instant shuttle call for most people, this puts these events on the same level of annoyance.

YakumoChen commented:
The events, as they are, are an annoyance, they're not supposed to be deadly or obnoxious to a ridiculous degree. Why do we need to buff them up from 'annoyance' to 'shuttle-calling hazard'?

Iamgoofball commented:
It's almost like we have an entire department and 8 job slots dedicated to fixing damage

AnturK commented:
Cmon, finish this up

Iamgoofball commented:
Will do soon

optimumtact commented:
goof

Iamgoofball commented:
Fuck

GunHog commented:
Put effort into your memes, Goof!

octareenroon91 commented:
The MDs need those sugar-foam coma patients

Iamgoofball commented:
ill fix it up this weekend
